docs/rabbbitmq.md

```diff
@@ -0,0 +1,91 @@
+| `password`              | required |                                 | The password used to connect to the server. |
+| `insecure`              | optional | true                            | Signals whether to use a secure TLS connection or not. If insecure is true TLS will not be enabled. |
+| `insecure_skip_verify`  | optional |                                 | Whether to skip verifying the certificate or not. A false value of insecure_skip_verify will not be used if insecure is true as the connection will not use TLS at all. |
+| `cert_file`             | optional |                                 | Path to the TLS cert to use for TLS required connections. |
+| `key_file`              | optional |                                 | Path to the TLS key to use for TLS required connections. |
+| `ca_file`               | optional |                                 | Path to the CA cert. As a client this verifies the server certificate. If empty, uses system root CA. |
+
+Example Configuration:
+
+
+```yaml
+metrics:
+  receivers:
+    rabbitmq:
+      type: rabbitmq 
+      endpoint: http://localhost:15672
+      password: pwd
+      username: usr
+  service:
+    pipelines:
+      rabbitmq:
+        receivers:
+          - rabbitmq
+```
+
+TCP connection with a username and password and TLS:
+
+```yaml
+metrics:
+  receivers:
+    rabbitmq:
+      type: rabbitmq 
+      endpoint: http://localhost:15672
+      collection_interval: 60s
+      password: pwd
+      username: usr
+      insecure: false
+      insecure_skip_verify: false
+      cert_file: /path/to/cert
+      ca_file: /path/to/ca
+  service:
+    pipelines:
+      rabbitmq:
+        receivers:
+          - rabbitmq
+```
+
+## Metrics
+
+The Ops Agent collects the following metrics from your rabbitmq instances.
+
+| Metric                                                 | Data Type | Unit        | Labels                          | Description    |
+| ---                                                    | ---       | ---         | ---                             | ---            | git 
+| rabbitmq.consumer.count | gauge | {consumers} | <ul> </ul>  | The number of consumers currently reading from the queue. |
+| rabbitmq.message.acknowledged | cumulative | {messages} | <ul> </ul>  | The number of messages acknowledged by consumers. |
+| rabbitmq.message.current | gauge | {messages} | <ul> <li>message.state</li> </ul>  | The total number of messages currently in the queue. |
+| rabbitmq.message.delivered | cumulative | {messages} | <ul> </ul>  | The number of messages delivered to consumers. |
+| rabbitmq.message.dropped | cumulative | {messages} | <ul> </ul>  | The number of messages dropped as unroutable. |
+| rabbitmq.message.published | cumulative | {messages} | <ul> </ul>  | The number of messages published to a queue. |
+
+## Attributes
+
+| Name | Description |
+| ---- | ----------- |
+| message.state | The state of messages in a queue. |
```

Comment:
How does the `attributes` table map to the [public documentation](https://cloud.google.com/monitoring/api/metrics_opsagent#3p)? Are the `xx.xx.xx` fields supposed to map to the labels? Should they be `state` instead without the `message.` prefix? And rename `attribute` to `labels`?

Reply:
yes that makes sense but state was used in different ways through out the metrics so I thought prefixing with the metric name would help differentiate between them. maybe an additional column for metric name would be useful in this table?

Reply:
Fixed, was a mis translation from otel.